pm.pl

```diff
@@ -1,25 +1,35 @@
 :- use_module(download).
 :- use_module(config).
-
+:- use_module(find_modules).
 create_dir_if_needed :- \+ exists_directory("deps"),
      make_directory("deps").
      
-download_to_disk(URL, Version) :- 
+download_to_disk(URL, Version, PackageName) :- 
     split_string(URL, "/", "", Pieces),
     last(Pieces, PackageName),
     format(atom(Formatted), 'deps/~w', [PackageName]),
     down(URL, Formatted).   
 
 sync :- create_dir_if_needed,
-        forall(dependency(URL, Version), download_to_disk(URL, Version)).
+        forall(dependency(URL, Version), download_to_disk(URL, Version, _)).
+
+handle_module_sources(File) :-
+                               format(atom(Formatted), 'deps/~w', [File]),
+                               find_deps(Formatted, [(Module, _)]),
+                               open('deps/sources.pl', append, Handle),
+                               format(atom(SourceLine), 'source("~w", "~w").', [Module, File]),
+                               writeln(Handle, SourceLine),
```

Comment:
replace the previous two lines with https://www.swi-prolog.org/pldoc/doc_for?object=write/2

Reply:
Actually writeln is fine, but you can write the term directly rather than converting it into some atom with the format call